ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/PruningResult.scala

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.v1
+
+import io.grpc.Status
+
+sealed trait PruningResult extends Product with Serializable
+
+/**
+  * Pruning has been performed. Commits the ledger api server to prune at the same offset passed to the
+  * WriteService.
+  */
+case object ParticipantPruned extends PruningResult
```

Comment:
Purely cosmetic: You could wrap the two cases in a `object PruningResult`, to allow users to write `PruningResult. ParticipantPruned`. We use the same pattern for [state updates](https://github.com/digital-asset/daml/blob/master/ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Update.scala#L29)